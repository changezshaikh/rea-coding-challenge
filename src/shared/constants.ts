export const PROPERTY_TYPES = {
  RESULTS: "results",
  SAVED: "saved",
};
