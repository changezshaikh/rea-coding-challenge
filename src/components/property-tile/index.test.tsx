import React from "react";
import { render } from "@testing-library/react";
import PropertyTile from ".";
import { PROPERTY_TYPES } from "../../shared/constants";

const testProperty = {
  price: "$726,500",
  agency: {
    brandingColors: {
      primary: "#ffe512",
    },
    logo:
      "https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif",
  },
  id: "1",
  mainImage:
    "http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg",
};

describe("Property Tile tests", () => {
  test("should render without crashing and match snapshot", () => {
    const { container } = render(
      <PropertyTile
        property={testProperty}
        propertyType={PROPERTY_TYPES.RESULTS}
        saveProperty={() => {}}
        removeProperty={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
