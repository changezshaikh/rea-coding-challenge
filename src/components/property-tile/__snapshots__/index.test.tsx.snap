// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Tile tests should render without crashing and match snapshot 1`] = `
<div>
  <div
    class="property__container "
  >
    <div
      class="property-header"
      style="background-color: rgb(255, 229, 18);"
    >
      <img
        alt="Agency Logo"
        class="logo-image"
        src="https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif"
      />
    </div>
    <img
      alt="Property"
      class="property-image"
      src="http://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg"
    />
    <div
      class="property-footer"
    >
      <span
        class="property-price"
      >
        $726,500
      </span>
      <button
        class="property-button "
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
